### ISSUE TITLE
Single quotes not recognised by export transformations and spellchecker

### ISSUE THREAD

**Issue by mxamber** [olivierkes/manuskript#929]
I'm not quite sure if these are one or two issues, but I would wager they have the same cause.

The export dialog allows replacing quotes. It can transform `'test'` into `‘test’` or into `‹test›` (a variant of quotation marks called _guillemet_ that's popular in some countries). Unfortunately, it also turns `'can't'` into `‹can›t›`. (It also fails to recognise the `'` at the end of a line as closing quotation mark).

A similar issue concerns the spellchecker: while double air quotes are correctly recognised as punctuation, single quotes are not. Therefore, if a sentence starts with `'`, the spellchecker will complain.

`'Come on', he spat. 'The master wants to see you.'`

Here, the spellchecker complains that `'Come` and `'The` aren't correct.

![image](https://user-images.githubusercontent.com/13611718/131555064-1a3aa6b6-a27f-42c1-ab35-b1402eb1d82f.png)

![image](https://user-images.githubusercontent.com/13611718/131555186-c6206acb-f780-4be2-a18d-2c7c123e3b6b.png)

**Comment by TheJackiMonster** [comment]
Yes, that's an issue. The current algorithm to find quotes to replace them is very naive. It just picks the next quote after another to complete a quotation but I think it should search until it finds one with a separator (,) or white space ( ) after it. But that's just an idea. I'm not sure if this would fulfill all cases correctly.

What you can do at the moment is using double-quotes (") for the sentences and replacing them later on with custom quotations like the ones you are using here for single-quotes. The exporter should allow this.

**Comment by mxamber** [comment]
If it's any help, these are the RegEx I put in the custom transformations for the time being. They're not perfect, probably, but so far, I haven't run into any issues.

Opening quotes: `([\s\n\b])(')([\w.])` ->` \1‘\3`

Closing quotes: `([\w.!?,])(')([\s\n,.!?])` -> `\1’\3`

Preserve apostrophes for contractions: `(\w)(['’])(\w)` -> `\1'\3`

**Comment by TheJackiMonster** [comment]
@mxamber Thanks, that should help a lot.

**Comment by TheJackiMonster** [comment]
The only problem I still see is that non-alphanumeric characters are not counted as `\w` which still messes with `ä`, `ö`, `ü` or similar cases from different languages. Otherwise RegEx make sense. I think we could either list all edge cases or look for another criteria to separate. Maybe we should only differentiate between white-space and non-white-space.
